src/engraving/rendering/dev/layoutindependent.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2023 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "layoutindependent.h"
+
+#include "global/async/asyncable.h"
+#include "draw/fontmetrics.h"
+
+#include "../../types/typesconv.h"
+#include "../../infrastructure/rtti.h"
+
+#include "../../dom/note.h"
+
+using namespace mu::draw;
+using namespace mu::engraving;
+using namespace mu::engraving::rendering::dev;
+using namespace mu::engraving::rtti;
+
+bool LayoutIndependent::isItemIndepended(const EngravingItem* item)
```

Comment:
I would not like to treat independent items very differently from other leaf items that do have a few dependencies. So, I wouldn't put the layout methods for these in a separate class.

We should not say:
"For independent items, we can calculate their complete layout immediately at the start."

We should say:
"For leaf items, we can calculate their graphical details (e.g. shape, bbox, and to some extent the positioning of subitems, like the wiggles of an arpeggio) whenever all data on which they depend has been calculated. (And if such item has no dependencies, that is incidentally immediately at the start)."